test/test_common/network_utility.cc

```diff
@@ -104,6 +104,11 @@ const std::string addressVersionAsString(const Address::IpVersion version) {
 
 Address::InstanceConstSharedPtr getSomeLoopbackAddress(Address::IpVersion version) {
   if (version == Address::IpVersion::v4) {
+#ifdef __APPLE__
```

Comment:
@jamessynge @htuch @hennna is there any point in really doing the random address thing for tests? Could we just delete that part and use canonical to remove ifdef?